test/TestExtensions/MockStorageProvider.cs

```diff
@@ -13,8 +14,27 @@
 namespace UnitTests.StorageTests
 {
     [DebuggerDisplay("MockStorageProvider:{Name}")]
-    public class MockStorageProvider : MarshalByRefObject, IStorageProvider
+    public class MockStorageProvider : IControllable, IStorageProvider
     {
+        public enum Commands
+        {
+            InitCount,
+            SetValue,
+            GetProvideState,
+            SetErrorInjection,
+            GetLastState,
+            ResetHistory
+        }
+        [Serializable]
+        public class StateForTest 
+        {
+            public int InitCount { get; set; }
+            public int CloseCount { get; set; }
+            public int ReadCount { get; set; }
+            public int WriteCount { get; set; }
+            public int DeleteCount { get; set; }
+        }
```

Comment:
This probably suppersedes the individual properties for InitCount, etc. They could be removed...